types/test/test-v8n.ts

```diff
@@ -0,0 +1,57 @@
+import v8n from '../index';
+
+v8n()
+  .not.equal('test')
+  .exact('test')
+  .number()
+  .integer()
+  .numeric()
+  .string()
+  .some.boolean()
+  .undefined()
+  .null()
+  .array()
+  .object()
+  .instanceOf(Error)
+  .pattern(/[a-z]/)
+  .every.lowercase()
+  .uppercase()
+  .vowel()
+  .consonant()
+  .first('t')
+  .last('t')
+  .empty()
+  .length(4)
+  .minLength(4)
+  .maxLength(4)
+  .negative()
+  .positive()
+  .between(4, 4)
+  .range(4, 4)
+  .lessThan(10)
+  .lessThanOrEqual(10)
+  .greaterThan(10)
+  .greaterThanOrEqual(10)
+  .even()
+  .odd()
+  .includes('test')
+  .schema({ test: 2 })
+  .passesAnyOf()
+  .optional(v8n().string())
+  .test(12);
+
+v8n()
+  .string()
+  .test();
+
+v8n()
+  .string()
+  .testAll();
+
+v8n()
+  .string()
+  .check();
+
+v8n()
+  .string()
+  .testAsync();
```

Comment:
What do you think of testing the return types as well?

Reply:
That is more of a "real test". Currently the tests are only compile-time. This would resolve itself when `v8n.js` is rewritten in actual TypeScript. As it stands at compile time only the types we provide (`.d.ts` definition) are available. Not sure that's really necessary for this "kinda" TypeScript integration.